### PR TITLE
feat: allow templates to specify max_ttl or autostop_requirement

### LINKED ISSUE #9284: quiet hours: deprecate and eventually remove max_ttl
We should only have one scheduling system in the product. We should deprecate and eventually remove the old max TTL-based approach to enforcing workspace shutdown. This should include some sort of warning in the UI and CLI. 

- [ ] Add migration that denotes which of `max_ttl` or `autostop_requirement` is used for each template 
- [ ] Update autostop code to use new table value
- [ ] Update UI to allow one of `max_ttl` or `autostop_requirement` (deprecate `max_ttl`)

### PR DESCRIPTION
**Pull request by deansheather**
- Allow both `max_ttl` and `autostop_requirement` at the same time in deployments
- Default to `CRON_TZ=UTC 0 0 * * *` for quiet hours schedule (so it's always enabled if entitled, can be overridden with flag still)
- Removes feature flag `template_autostop_requirement`, merged into `advanced_template_scheduling`

## TODO:

- [x] API for template to select which one to use
- [x] Show both in UI, only let one be set at a time

<details>
  <summary>Create template form</summary>
  
https://github.com/coder/coder/assets/11241812/4dc02c41-275b-45c2-a1fe-f776aa2126bb

</details>

<details>
  <summary>Edit template form</summary>
  

https://github.com/coder/coder/assets/11241812/c715d914-b83a-4ab7-a8af-7e031f3080ba

</details>

Closes #9284

### CODE REVIEW

File: site/src/components/Form/Form.tsx [hunk 1]
```diff
@@ -121,6 +122,7 @@ export const FormSection: FC<
         >
           {title}
           {alpha && <AlphaBadge />}
+          {deprecated && <DeprecatedBadge />}
```
Comment (Emyrk) [inline site/src/components/Form/Form.tsx:125]: Can you drop a screencap with what this looks like?

**Review by Emyrk — APPROVED** [review]
BE is good, just need to remove a line to make tests work

File: site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateScheduleForm.tsx [hunk 2]
```diff
@@ -338,85 +358,115 @@ export const TemplateScheduleForm: FC<TemplateScheduleForm> = ({
...
+            label="Weeks between required stops"
+            type="number"
+          />
         </Stack>
       </FormSection>
 
-      {allowAutostopRequirement && (
-        <FormSection
-          title="Autostop Requirement"
-          description="Define when workspaces created from this template are stopped periodically to enforce template updates and ensure idle workspaces are stopped."
```
Comment (aslilac) [inline site/src/pages/TemplateSettingsPage/TemplateSchedulePage/TemplateScheduleForm.tsx:368]: thank you for rewording this 🙌

File: coderd/schedule/template.go [hunk 3]
```diff
@@ -117,14 +117,12 @@ type TemplateScheduleOptions struct {
...
-	// default
-	MaxTTL time.Duration `json:"max_ttl"`
-	// UseAutostopRequirement dictates whether the autostop requirement should
-	// be used instead of MaxTTL. This is governed by the feature flag and
-	// licensing.
+	MaxTTL               time.Duration `json:"max_ttl"`
+	// UseMaxTTL dictates whether the max_ttl should be used instead of
+	// autostop_requirement for this template. This is governed by the template
+	// and licensing.
 	// TODO(@dean): remove this when we remove max_tll
```
[the remark this answers is not in the document]
Comment (deansheather) [reply coderd/schedule/template.go:124]: Yes, it's deprecated

File: site/src/components/Form/Form.tsx (hunk 1 above)
Comment (deansheather) [reply site/src/components/Form/Form.tsx:125]: image

**Pull request merged**